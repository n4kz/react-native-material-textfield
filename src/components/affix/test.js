import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Affix from '.';

/* eslint-env jest */

const props = {
  fontSize: 16,
  baseColor: 'blue',
  animationDuration: 225,
};

const prefix = 'a';
const suffix = 'z';

it('renders prefix', () => {
  let affix = renderer
    .create(<Affix type='prefix' {...props}>{prefix}</Affix>)
    .toJSON();

  expect(affix)
    .toMatchSnapshot();
});

it('renders suffix', () => {
  let affix = renderer
    .create(<Affix type='suffix' {...props}>{suffix}</Affix>)
    .toJSON();

  expect(affix)
    .toMatchSnapshot();
});
