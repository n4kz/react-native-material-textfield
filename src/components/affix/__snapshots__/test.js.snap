// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders prefix 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "alignSelf": "flex-start",
      "height": 24,
      "justifyContent": "center",
      "opacity": 0,
      "paddingRight": 8,
      "top": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "blue",
        "fontSize": 16,
        "textAlign": "left",
      }
    }
  >
    a
  </Text>
</View>
`;

exports[`renders suffix 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "alignSelf": "flex-start",
      "height": 24,
      "justifyContent": "center",
      "opacity": 0,
      "paddingLeft": 8,
      "top": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "blue",
        "fontSize": 16,
        "textAlign": "right",
      }
    }
  >
    z
  </Text>
</View>
`;
